core/include/detray/masks/line.hpp

```diff
@@ -150,14 +152,42 @@ class line {
               template <typename, std::size_t> class bounds_t,
               typename scalar_t, std::size_t kDIM,
               typename std::enable_if_t<kDIM == e_size, bool> = true>
-    DETRAY_HOST_DEVICE inline std::array<scalar_t, 6> local_min_bounds(
-        const bounds_t<scalar_t, kDIM>& bounds,
+    DETRAY_HOST_DEVICE inline darray<scalar_t, 6> local_min_bounds(
+        const bounds_t<scalar_t, kDIM> &bounds,
         const scalar_t env = std::numeric_limits<scalar_t>::epsilon()) const {
         assert(env > 0.f);
         const scalar_t xy_bound{bounds[e_cross_section] + env};
         const scalar_t z_bound{bounds[e_half_z] + env};
         return {-xy_bound, -xy_bound, -z_bound, xy_bound, xy_bound, z_bound};
     }
+
+    /// @brief Check consistency of boundary values.
+    ///
+    /// @param bounds the boundary values for this shape
+    /// @param os output stream for error messages
+    ///
+    /// @return true if the bounds are consistent.
+    template <template <typename, std::size_t> class bounds_t,
+              typename scalar_t, std::size_t kDIM,
+              typename std::enable_if_t<kDIM == e_size, bool> = true>
+    DETRAY_HOST constexpr bool check_consistency(
+        const bounds_t<scalar_t, kDIM> &bounds, std::ostream &os) const {
+
+        constexpr auto epsilon{10.f * std::numeric_limits<scalar_t>::epsilon()};
```

Comment:
Why do you multiply 10 here?

Reply:
Because ```std::numeric_limits<scalar_t>::epsilon()``` is the minimal difference between two floating point numbers, which I thought is too small for a tolerance on mask values. Let me rename the tolerance